File: java/src/main/java/ai/pagou/examples/lib/Format.java
package ai.pagou.examples.lib;

import java.text.NumberFormat;
import java.util.Currency;
import java.util.Locale;

/** Small presentation helpers shared by the runnable examples. */
public final class Format {

  private static final Locale PT_BR = Locale.forLanguageTag("pt-BR");

  private Format() {}

  /** Formats an integer amount in the smallest currency unit as a display string. */
  public static String amount(long cents, String currency) {
    NumberFormat formatter = NumberFormat.getCurrencyInstance(PT_BR);
    try {
      formatter.setCurrency(Currency.getInstance(currency));
    } catch (IllegalArgumentException ignored) {
      // Unknown currency code: fall back to the locale default.
    }
    return formatter.format(cents / 100.0);
  }

  public static String amount(long cents) {
    return amount(cents, "BRL");
  }

  /** A short, unique idempotency key for a given operation and reference. */
  public static String idempotencyKey(String operation, String reference) {
    return operation + "_" + reference;
  }

  /** Prints a labelled JSON block for readable script output. */
  public static void printResult(String label, Object value) {
    System.out.println();
    System.out.println(label + ":");
    System.out.println(Json.pretty(value));
  }
}

File: php/src/Format.php
<?php

declare(strict_types=1);

namespace Pagou\Examples;

use NumberFormatter;

final class Format
{
    /** Formats an integer amount in the smallest currency unit as a display string. */
    public static function amount(int $cents, string $currency = 'BRL'): string
    {
        if (class_exists(NumberFormatter::class)) {
            $formatter = new NumberFormatter('pt_BR', NumberFormatter::CURRENCY);
            return (string) $formatter->formatCurrency($cents / 100, $currency);
        }
        return $currency . ' ' . number_format($cents / 100, 2, ',', '.');
    }

    /** A short, unique idempotency key for a given operation and reference. */
    public static function idempotencyKey(string $operation, string $reference): string
    {
        return "{$operation}_{$reference}";
    }

    /** Prints a labelled JSON block for readable script output. */
    public static function printResult(string $label, mixed $value): void
    {
        echo PHP_EOL . $label . ':' . PHP_EOL;
        echo json_encode($value, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE) . PHP_EOL;
    }
}
